pkg/proto/agentbaker/v1/config.proto

```diff
@@ -0,0 +1,225 @@
+  string resource_group = 9;
+  string location = 10;
+  string vm_type = 11;  
+  
+  string primary_availability_set = 12;
+  string primary_scale_set = 13;
+  
+  CloudProviderConfig cloud_provider_config = 14;
+  IdentityConfiguration identity_configuration = 15;
+
+  RuncConfig runc_config = 16;
+
+  // ... and a bunch of user-specified config
+  string repo_depot_endpoint = 17;
+  string linux_admin_username = 18;
+    // removed mobyVersion
+  string tenant_id = 19;
+  bool use_instance_metadata = 20;
+  LoadBalancerConfig load_balancer_config = 21;
+    // removed containerRuntime
+    // removed cliTool because docker is deprecated. Only "ctr" will be supported.
+  ContainerdConfig containerd_config = 22;
+  bool is_vhd = 23;
+  GpuConfig gpu_config = 24;
+  bool is_sgx_node = 25;
+  TeleportConfig teleport_config = 26;
+  // removed NEEDS_CONTAINERD because it's always needed now
+
+  HTTPProxyConfig http_proxy_configuration = 27;
+  repeated string trusted_certificate_authorities = 28;
+  FeatureState ssh_status = 29;
+  FeatureState unattended_upgrade_status = 30;
+  string message_of_the_day = 31;
+  string swap_file_size_mb = 32;
+  SysctlConfig sysctls = 33;
+  KubeletConfiguration kubelet_configuration = 34;
+  optional string gpu_instance_profile = 35;
+  FeatureState hosts_config_agent_status = 36;
+}
+
+message KubeletConfiguration {
+  // TODO(ace): remove these/make api defensible
+  repeated string kubelet_flags = 1;
+  map<string, string> kubelet_node_labels = 2;
+  repeated Taint taints = 3;
+  repeated Taint startup_taints = 4;
+  KubeletDiskType kubelet_disk_type = 5;
+}
+
+message SysctlConfig {
+  repeated Sysctl sysctls = 1;
+}
+
+enum FeatureState {
+  FEATURE_STATE_UNSPECIFIED = 0;
+  FEATURE_STATE_ENABLED = 1;
+  FEATURE_STATE_DISABLED = 2;
+}
+
+message IdentityConfiguration {
+  IdentityType identity_type = 1; // Q: I am assuming this is a better pattern than Oneof?
+  string service_principal_id = 2;
+  string service_principal_secret = 3;
+  string assigned_identity_id = 4;  //could be user or system assigned, depending on the type
+}
+
+enum IdentityType {
+  IDENTITY_TYPE_UNSPECIFIED = 0;
+  IDENTITY_TYPE_SERVICE_PRINCIPAL = 1;
+  IDENTITY_TYPE_SYSTEM_IDENTITY = 2;
+  IDENTITY_TYPE_USER_IDENTITY = 3;
+}
+
+enum NetworkPluginType {
+  NETWORK_PLUGIN_TYPE_UNSPECIFIED = 0;
+  NETWORK_PLUGIN_TYPE_NONE = 1;
+  NETWORK_PLUGIN_TYPE_AZURE = 2;
+  NETWORK_PLUGIN_TYPE_KUBENET = 3;
+}
+
+enum NetworkPolicyType {
+  NETWORK_POLICY_TYPE_UNSPECIFIED = 0;
+  NETWORK_POLICY_TYPE_NONE = 1;
+  NETWORK_POLICY_TYPE_AZURE = 2;
+  NETWORK_POLICY_TYPE_CALICO = 3;
+}
+
+message HTTPProxyConfig {
+  optional string http_proxy = 1;
+  optional string https_proxy = 2;
+  repeated string no_proxy = 3;
+  optional string trusted_certificate_authority = 4;
+  FeatureState http_proxy_config_status = 5;
+  FeatureState http_proxy_ca_config_status = 6;
+  string http_proxy_trusted_ca = 7;
+  FeatureState custom_ca_trust_status = 8;
```

Comment:
I think these should be separate from HTTPProxyConfig since custom ca trust is a different configuration, will also let @mikolaj weigh in here.

Reply:
Makes sense. updated

Reply:
Yeah please keep both. They are two separate features, albeit similar :D